sandman/source/logger.inl

```diff
@@ -0,0 +1,146 @@
+#include "logger.h"
+
+template <typename T, typename... ParamsT>
+[[gnu::always_inline]] inline void ::Logger::Write(T&& firstArg, ParamsT&&... args)
+{
+	static_assert(not std::is_same_v<std::decay_t<T>, Shell::Attr>,
+					  "Do not pass in attributes directly; instead use an attribute object wrapper.");
+
+	if constexpr (Logger::IsFormat<std::decay_t<T>>)
+	{
+		// If the first argument is a `Logger::Format` object,
+		// then forward the arguments to a call to `Logger::FormatWrite`.
+		std::apply
+		(
+			[this, formatString=firstArg.m_FormatString](auto&&... objects) -> void
+			{
+				return this->FormatWrite(formatString, std::forward<decltype(objects)>(objects)...);
+			},
+			firstArg.m_Objects
+		);
+	}
+	else if constexpr (Shell::IsAttrWrapper<std::decay_t<T>>)
+	{
+		// Callable to be passed into `std::apply`.
+		auto const writeArgs = [this](auto&&... objects) -> void
+		{
+			return this->Write(std::forward<decltype(objects)>(objects)...);
+		};
+
+		if (m_ScreenEcho)
+		{
+			// Get the current data from the buffer.
+			std::string const string(m_Buffer.str());
+
+			// Dump the current data to the output destinations.
+			bool const didPushAttributes
```

Comment:
I prefer list initialization for primitive types because it prevents narrowing conversions.

```C++
/* This compiles. */

// implicit conversion to `true`
bool booleanValue = 2;

// underflows to 4294967295
unsigned int unsignedInteger = -1;
```

```C++
/* This compiles. */

// implicit conversion to `true`
bool booleanValue(2);

// underflows to 4294967295
unsigned int unsignedInteger(-1);
```

```C++
/* This does not compile. */

// "error: narrowing conversion of ‘2’ from ‘int’ to ‘bool’"
bool booleanValue{2};

// "error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’"
unsigned int unsignedInteger{-1};
```